pvlib/tests/test_singlediode.py

```diff
@@ -408,3 +408,112 @@ def test_pvsyst_breakdown(method, brk_params, recomb_params, poa, temp_cell,
 
     vsc_88 = bishop88_v_from_i(isc_88, *x, **y, method=method)
     assert np.isclose(vsc_88, 0.0, *tol)
+
+
+def bishop88_arguments():
+    pvsyst_fs_495 = get_pvsyst_fs_495()
+    # evaluate PVSyst model with thin-film recombination loss current
+    # at reference conditions
+    x = pvsystem.calcparams_pvsyst(
+        effective_irradiance=pvsyst_fs_495['irrad_ref'],
+        temp_cell=pvsyst_fs_495['temp_ref'],
+        alpha_sc=pvsyst_fs_495['alpha_sc'],
+        gamma_ref=pvsyst_fs_495['gamma_ref'],
+        mu_gamma=pvsyst_fs_495['mu_gamma'], I_L_ref=pvsyst_fs_495['I_L_ref'],
+        I_o_ref=pvsyst_fs_495['I_o_ref'], R_sh_ref=pvsyst_fs_495['R_sh_ref'],
+        R_sh_0=pvsyst_fs_495['R_sh_0'], R_sh_exp=pvsyst_fs_495['R_sh_exp'],
+        R_s=pvsyst_fs_495['R_s'],
+        cells_in_series=pvsyst_fs_495['cells_in_series'],
+        EgRef=pvsyst_fs_495['EgRef']
+    )
+    y = dict(d2mutau=pvsyst_fs_495['d2mutau'],
+             NsVbi=VOLTAGE_BUILTIN*pvsyst_fs_495['cells_in_series'])
+    # Convert (*x, **y) in a bishop88_.* call to dict of arguments
+    args_dict = {
+        'photocurrent': x[0],
+        'saturation_current': x[1],
+        'resistance_series': x[2],
+        'resistance_shunt': x[3],
+        'nNsVth': x[4],
+    }
+    args_dict.update(y)
+    return args_dict
+
+
+@pytest.mark.parametrize('method, kwargs', [
+    ('newton', {
+        'tol': 1e-3,
+        'rtol': 1e-3,
+        'maxiter': 30,
+    }),
+    ('brentq', {
+        'xtol': 1e-3,
+        'rtol': 1e-3,
+        'maxiter': 30,
+    })
+])
+def test_bishop88_kwargs_pass(method, kwargs):
```

Comment:
I think this test is technically not quite valid.  The default tolerance of 1e-6 is stricter than the 1e-3 here, so if there was a bug where a user-supplied tolerance didn't actually get passed through to scipy (and the default of 1e-6 was used instead), I think these tests would still pass.

I think it would be better to not base the checks on numerical results anyway though.  Really all we want to check is that the kwarg was passed to scipy correctly.  For that we can use `mocker.spy` and `assert_called_once_with` as described here: https://pvlib-python.readthedocs.io/en/stable/contributing.html#pvsystem-location

Reply:
Okay, didn't know of that way of doing tests, and I like it.
I will check those tools and work on it these days. I can't tell when I will have finished.

Reply:
I had to make use of other methods to check the calls were right. Could you please have a look at them?